src/engraving/rendering/dev/chordlayout.cpp

```diff
@@ -2101,10 +2123,19 @@ double ChordLayout::layoutChords2(std::vector<Note*>& notes, bool up, LayoutCont
 
     for (int idx = startIdx; idx != endIdx; idx += incIdx) {
         Note* note    = notes[idx];                         // current note
-        int line      = note->line();                       // line of current note
+        const int line      = note->stringOrLine();                       // line of current note
         Chord* chord  = note->chord();
-        staff_idx_t staffIdx  = chord->vStaffIdx();                 // staff of current note
-        track_idx_t trackIdx = chord->track();
+
+        const staff_idx_t staffIdx  = chord->vStaffIdx();                 // staff of current note
+        const track_idx_t trackIdx = chord->track();
+        const Staff* st = note->staff();
+        const StaffType* tab = st->staffTypeForElement(note);
+        const bool isTab = note->staff() && note->staff()->isTabStaff(note->chord()->tick());
+
+        if (isTab) {
+            // Need to lay out TAB note to know width
+            TLayout::layoutNote(note, note->mutldata());
```

Comment:
Similar question as before: why do we need this here for TABs but not for normal staves? Ideally, this should be done in the same place as it's done for normal staves (that is, I would imagine, well before we get here)

Reply:
The layout in this area just uses the notehead symbol dimensions and doesn't need the whole note to be laid out.  
TAB notes need to have their `m_fretString` correctly (which happens during note layout) before `tabHeadWidth` can return an accurate value.  I can change `tabHeadWidth` to set `m_fretString` to avoid needing to call note layout.

Reply:
Ah ok I see, makes sense. Then I'd leave it as it is (perhaps, again, with a comment to give this context). Let's keep tabHeadWidth single-responsibility and do the layout call here